staging/src/k8s.io/code-generator/cmd/validation-gen/validators/validators.go

```diff
@@ -21,9 +21,36 @@ import "k8s.io/gengo/v2/types"
 // DeclarativeValidator is able to extract validation function generators from
 // types.go files.
 type DeclarativeValidator interface {
-	// ExtractValidations returns a FunctionGen for each validation this DeclarativeValidator
+	// ExtractValidations returns a ValidatorGen for the validation this DeclarativeValidator
 	// supports for the given go type, and it's corresponding comment strings.
-	ExtractValidations(t *types.Type, comments []string) ([]FunctionGen, error)
+	ExtractValidations(t *types.Type, comments []string) (ValidatorGen, error)
+}
+
+// ValidatorGen defines the function calls and variables to generate to perform validation.
+type ValidatorGen struct {
```

Comment:
Super nit - should this be called `Validations` now?  I think it would read better.

Reply:
SGTM!